code/game/turfs/simulated/floor/light_floor.dm

```diff
@@ -10,23 +10,51 @@
 	var/light_broken = FALSE
 	/// Can we modify a colour
 	var/can_modify_colour = TRUE
+	/// Are we draining power, prevents updating the icon drain / generate power
+	var/using_power = FALSE
 
 /turf/simulated/floor/light/Initialize(mapload)
 	. = ..()
 	update_icon()
+	var/area/current_area = get_area(src)
+	var/obj/machinery/power/apc/current_apc = current_area.get_apc()
+	if(current_apc)
+		RegisterSignal(current_area, COMSIG_AREA_POWER_CHANGE, PROC_REF(power_update), override = TRUE)
 
 /turf/simulated/floor/light/update_icon_state()
+	var/area/A = get_area(src)
+	if(!A)
+		return
 	if(on)
 		icon_state = "light_on"
 		set_light(5, null, color)
+		if(!using_power)
+			A.addStaticPower(100, STATIC_LIGHT)
+			using_power = TRUE
 	else
 		icon_state = "light_off"
 		set_light(0)
+		if(using_power)
+			A.addStaticPower(-100, STATIC_LIGHT)
+			using_power = FALSE
+
```

Comment:
not 100% happy with non-icon related logic being in an update_icon_state. I would appreciate it if you did a minor refactor and move this logic to `toggle_light`

Reply:
done